### PR TITLE
Add pagination for petitions (#240)

### LINKED ISSUE #240: Add pagination for petitions
Hello there!

**Is your feature request related to a problem? Please describe.**
I find that the user experience lacks of fluidity when I want to access all the petitions. I first see only the top 12 then I need to access the page with all petitions which will start with the same 12 petitions.

It is also not very clear when there is fewer than 12 petitions, as the button "show all petitions" still appear and is basically pointless 

**Describe the solution you'd like**
I think it would be more fluid to add pagination directly on the home page, with 12 petitions per page to keep the same interface as currently.

**Additional context**
It's really easy to do so in Django, I can take care of this change, I just wanted to confirm first that this is something you want to be done :)

### PR DESCRIPTION
**Pull request by simpnu**
Fixes #240 

image

**Notes:**
* I added a simple pagination, without "first" and "last" buttons, this could be added later on
* I made it a way to avoid making changes which require new translations, but that's up to you
* I made the choice to add the pagination code into `generic_petition_list.html` rather than `index.html`, but it could work on either

### CODE REVIEW

File: pytition/petition/views.py [hunk 1]
```diff
@@ -74,6 +74,11 @@ def index(request):
             user = get_session_user(request)
         else:
             user = request.user
+        all_petitions = Petition.objects.filter(published=True).order_by('-id')
+        paginator = Paginator(all_petitions, 12)
```
Comment (martinlehoux) [inline pytition/petition/views.py:78]: Just one thing, i would put the `12` value either in a constant or in Django settings if it's supposed to be modifiable.

Comment (fallen) [reply pytition/petition/views.py:78]: ah right, good idea!
I'd go for the Django settings `in base.py`

Comment (simpnu) [reply pytition/petition/views.py:78]: Good idea, I'll add it to the base settings

**Comment by simpnu** [comment]
@fallen Damn, I totally forgot to check whether the template `generic_petition_list.html` was used for other views... Not sure neither how `petitions.has_other_pages` doesn't crash but it seems this case is handled by Django.

I'm not sure what is the best option here:
1.  Move the pagination HTML to index.html
2. Add the paginator everywhere

I'd go with option 1 personally, as I'm not sure pagination makes sense everywhere (for instance for the `all_petition` option).

WDYT?

**Comment by simpnu** [comment]
@fallen Then I guess it doesn't make sense to keep the whole `all_petitions.html` template, as it is essentially the same as `index.html` now with pagination.

I can make the changes. I'll add it anyway on org and user profile, it will only show if there are more than 12 petitions, otherwise it'll be the same as now

**Comment by fallen** [comment]
> @fallen Then I guess it doesn't make sense to keep the whole `all_petitions.html` template, as it is essentially the same as `index.html` now with pagination.
> 
> I can make the changes. I'll add it anyway on org and user profile, it will only show if there are more than 12 petitions, otherwise it'll be the same as now

Sure, you're right, I think we can get rid of `all_petitions.html`. Nice catch!

**Comment by simpnu** [comment]
Just made the changes as discussed.

* Use a setting to define the number of elements per page
* Update the tests to reflect the changes
* Keep a redirection from `/all_petitions` to make sure it doesn't break existing links
* I didn't add pagination for the search as it requires to add more complexity (to handle query parameters). I've already done it on another project, so I can do it if needed, but it requires to create a custom templatetag
* Note that my IDE removes trailing whitespaces, so don't be surprised. I can revert if needed

**Pull request merged**